appholder/src/main/java/com/ul/ims/gmdl/appholder/util/NfcHandler.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package com.ul.ims.gmdl.appholder.util
+
+import android.nfc.cardemulation.HostApduService
+import android.os.Bundle
+import android.util.Log
+import com.ul.ims.gmdl.appholder.transfer.TransferManager
+
+class NfcHandler : HostApduService() {
+
+    companion object {
+        private const val LOG_TAG = "NfcHandler"
+    }
+
+    override fun processCommandApdu(commandApdu: ByteArray, extras: Bundle?): ByteArray? {
+        Log.d(LOG_TAG, "processCommandApdu: Command-> ${FormatUtil.encodeToString(commandApdu)}")
+
+        TransferManager.getInstance(applicationContext)
+            .nfcEngagementProcessCommandApdu(this, commandApdu)
+
+        return null
+    }
+
+    override fun onDeactivated(reason: Int) {
+        Log.d(LOG_TAG, "onDeactivated: reason-> $reason")
+        TransferManager.getInstance(applicationContext).nfcEngagementOnDeactivated(this, reason)
+    }
+}
```

Comment:
I think you also need to add code to `MainActivity.kt` to call `NfcAdapter.enableForegroundDispatch()` and `NfcAdapter.disableForegroundDispatch()` the same way that the IC testapp does. Otherwise you may run into problems if you have multiple mDL applications installed...

Reply:
I enabled NFC foreground dispatch, but I could not see any difference in my test devices